### PR TITLE
Additional functional

### PR DESCRIPTION
**Pull request by yu-krugliak**

### CODE REVIEW

**Comment by yu-krugliak** [comment]
Please, see my pr for problems

**Comment by yu-krugliak** [comment]
In this PR I added remaining functions for Epic1, it's:
1. Assign genres to game
2. Apply filter for the list of the games ( by ‘genre’, by ‘name’)
3. Tie am image to game

File: OnlineGameStore.Api/OnlineGameStore.Api.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="CloudinaryDotNet" Version="1.20.0" />
```
Comment (yu-krugliak) [inline OnlineGameStore.Api/OnlineGameStore.Api.csproj:10]: Remove unused dependency

File: OnlineGameStore.Api/StartupExtensions/StartupApplication.cs [hunk 2]
```diff
@@ -1,4 +1,7 @@
-﻿using MapsterMapper;
```
Comment (yu-krugliak) [inline OnlineGameStore.Api/StartupExtensions/StartupApplication.cs:1]: Remove unused namespaces in entire solution

File: OnlineGameStore.Application/Services/Implementation/GameService.cs [hunk 3]
```diff
@@ -81,25 +78,61 @@ public async Task DeleteByIdAsync(Guid gameId)
...
+            await AddPlatformsToGame(gameRequest.PlatformIds, game!);
+
+            var isUpdated = await _gameRepository.UpdateAsync(game!);
 
             if (!isUpdated)
             {
                 throw new ServerErrorException("Can't update this game.", null);
             }
-        }    
+        }
```
Comment (yu-krugliak) [inline OnlineGameStore.Application/Services/Implementation/GameService.cs:99]: Simplify logic to get rid of `UpdateGenresAsync()` method

File: OnlineGameStore.Application/Services/Implementation/StorageService.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            _cloudinary = new(account);
+        }
+
+        public async Task<string> UploadImageAsync(IFormFile image, string folder)
+        {
+            var result = await _cloudinary.UploadAsync(new ImageUploadParams
+            {
+                File = new FileDescription(image.FileName, image.OpenReadStream()),
+                Tags = Tags,
+                Folder = $"OnlineGameStoreMedia/{folder}"
```
Comment (yu-krugliak) [inline OnlineGameStore.Application/Services/Implementation/StorageService.cs:29]: Move OnlineGameStoreMedia to constants

File: OnlineGameStore.Infrastructure/Repositories/Implementations/GameRepository.cs [hunk 5]
```diff
@@ -32,26 +32,25 @@ public GameRepository(GamesContext gamesContext) : base(gamesContext)
...
-            return await _gamesContext.Games
+            var filtered = _gamesContext.Games
                 .Include(game => game.Genres)
-                .Where(game =>
-                    game.Genres!.Any(genre => genre.Id == genreId)
-                )
                 .Include(game => game.Platforms)
-                .ToListAsync();
-        }
+                .AsNoTracking();
```
Comment (yu-krugliak) [inline OnlineGameStore.Infrastructure/Repositories/Implementations/GameRepository.cs:40]: Add `AsNoTracking()` where it's necessary

**Review by ataromoku — APPROVED** [review]

**Pull request merged**